app/lib/actions/katello/repository/filtered_index_content.rb

```diff
@@ -21,16 +21,26 @@ def run
             ::Katello::FileUnit.import_for_repository(repo)
           elsif repo.deb?
             ::Katello::Deb.import_all(unit_ids, repo)
-          else
-            ::Katello::Rpm.import_all(unit_ids, repo)
+          elsif repo.yum?
+            if input[:content_type] == 'srpm'
+              ::Katello::Srpm.import_all(unit_ids, repo)
+            else
+              ::Katello::Rpm.import_all(unit_ids, repo)
+            end
           end
         end
 
         private
 
         def search_units(repo)
-          found = repo.unit_search(:type_ids => [repo.unit_type_id],
+          content_type = ::Katello::RepositoryTypeManager.find(repo.content_type).default_managed_content_type
```

Comment:
default_managed_content_type returns a class, we want to figure out the pulp2 unit type

Reply:
So i think you could just add ::CONTENT_TYPE onto the end of it

Reply:
actually, let me show you an alternative.